app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/NetworkChangeReceiver.java

```diff
@@ -0,0 +1,75 @@
+package com.teamagam.gimelgimel.app.network.receivers;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+
+import com.teamagam.gimelgimel.app.network.rest.RestAPI;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.Call;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+/**
+ * An implementation of {@link BroadcastReceiver}, that gets notification every time the network changes.
+ */
+public class NetworkChangeReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        final ConnectivityManager connectivityManager =
```

Comment:
Extract two methods out of this method.
isNetwrokAvailbale
broadcastConnectivityStatus !resolve